// Указываем синтаксис proto3
syntax = "proto3";

// Определяем пакет для нашего сервиса
package document_uploader;

// Сервис для загрузки документов
service DocumentUploader {
  // Client streaming RPC: клиент отправляет поток чанков файла
  rpc UploadFile(stream FileChunk) returns (UploadStatus) {}
}

// Сообщение-чанк: часть файла
message FileChunk {
  string file_name = 1; // Имя файла (передается, скорее всего, в первом чанке)
  bytes content = 2;     // Бинарное содержимое части файла
  uint32 chunk_size = 3; // Размер этого чанка (опционально, для диагностики)
}

// Сообщение-ответ с результатом загрузки
message UploadStatus {
  string message = 1; // Сообщение об успехе или ошибке
  bool success = 2;   // Флаг успешности
  string file_path = 3; // Путь, по которому был сохранен файл на сервере
}